staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -381,7 +391,7 @@ func (d *DiscoveryClient) ServerVersion() (*version.Info, error) {
 	var info version.Info
 	err = json.Unmarshal(body, &info)
 	if err != nil {
-		return nil, fmt.Errorf("got '%s': %v", string(body), err)
+		return nil, &ErrParseServerVersionFailed{Err: err}
```

Comment:
just `fmt.Errorf("unable to parse the server version: %s", err)`

Reply:
updated